core/src/main/scala/org/apache/spark/scheduler/cluster/SparkDeploySchedulerBackend.scala

```diff
@@ -137,7 +137,7 @@ private[spark] class SparkDeploySchedulerBackend(
 
   override def executorRemoved(fullId: String, message: String, exitStatus: Option[Int]) {
     val reason: ExecutorLossReason = exitStatus match {
-      case Some(code) => ExecutorExited(code, isNormalExit = true, message)
+      case Some(code) => ExecutorExited(code, isNormalExit = false, message)
```

Comment:
I don't think that's a good idea. The user's code could then `System.exit(0)` and mess up accounting of failed tasks.